core/include/irods/private/http_api/common.hpp

```diff
@@ -174,6 +184,13 @@ namespace irods
 	auto fail(boost::beast::error_code ec, char const* what) -> void;
 
 	auto enable_ticket(RcComm& _comm, const std::string& _ticket) -> int;
+
+	template <std::size_t N>
+	// NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+	constexpr auto strcpy_null_terminated(char (&_dst)[N], const char* _src) -> char*
```

Comment:
Should this be `stncpy_null_terminated`?

Also, can you explain why this is a template function rather than passing the size as a separate argument? To prevent possibly misusing it?

Reply:
> Should this be `stncpy_null_terminated`?

It originally had `strncpy` in the name, but I moved away from that because it's an implementation detail. The impl could change one day and not use `strncpy`, who knows.

> Also, can you explain why this is a template function rather than passing the size as a separate argument? To prevent possibly misusing it?

That's correct. This implementation can only be used with compile-time arrays.

Reply:
> It originally had `strncpy` in the name, but I moved away from that because it's an implementation detail. The impl could change one day and not use `strncpy`, who knows.

That sounds to me like an argument to remove the reference to the `strcpy` family altogether and use something like `safely_copy_string` or something like that. But the current name is fine.
 
> That's correct. This implementation can only be used with compile-time arrays.

Okay. I think I had to ask because seeing it triggered something in my brain from my previous job where generating a lot of functions like this would blow up the size of the compiled code too much to fit on the limited storage. :)

Reply:
> That sounds to me like an argument to remove the reference to the `strcpy` family altogether and use something like `safely_copy_string` or something like that. But the current name is fine.

True. Thinking of going with `copy_string`. Or maybe I'll go back to `strncpy_null_terminated`. Starting to think it's just easier to understand.

> Okay. I think I had to ask because seeing it triggered something in my brain from my previous job where generating a lot of functions like this would blow up the size of the compiled code too much to fit on the limited storage. :)

I've heard about that too, but I don't think we have to worry about that right now.

Reply:
Changed back to `strncpy_null_terminated`.